src/pages/requests/Create.tsx

```diff
@@ -188,18 +188,50 @@ const UNTIL_JUST_NUMERIC_ID_TO_LABELS: Record<string, string> = {
 
 const UNTIL_OPTIONS = Object.entries(UNTIL_ID_TO_LABELS).map(([id, label], index) => ({id: id, label: label}));
 
+function filterUntilLabels(timeLimit: number): [string, Array<Record<string, string>>] {
+  const filteredUntil = Object.keys(UNTIL_JUST_NUMERIC_ID_TO_LABELS)
+    .filter((key) => Number(key) <= timeLimit!)
+    .reduce(
+      (obj, key) => {
+        obj[key] = UNTIL_JUST_NUMERIC_ID_TO_LABELS[key];
+        return obj;
+      },
+      {} as Record<string, string>,
+    );
+
+  const filteredLabeles = Object.entries(Object.assign({}, filteredUntil, {custom: 'Custom'})).map(
+    ([id, label], index) => ({
+      id: id,
+      label: label,
+    }),
+  );
+
+  return [Object.keys(filteredUntil).at(-1)!, filteredLabeles];
+}
+
 function CreateRequestContainer(props: CreateRequestContainerProps) {
   const navigate = useNavigate();
 
-  const [until, setUntil] = React.useState('1209600');
+  const [timeLimit, setTimeLimit] = React.useState<number | null>(
```

Comment:
Can you add a comment here describing what is being calculated for the time limit here?